### ISSUE TITLE
matrix constructor: Accept "base_ring" keyword

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#33380]
<div id="comment:0"></div>

Right now, if not passed as a positional argument, it has to be provided as `ring`.

```
sage: matrix(5, base_ring=QQ)
TypeError: __classcall__() got multiple values for argument 'base_ring'
sage: matrix(5, ring=QQ)
[0 0 0 0 0]
[0 0 0 0 0]
[0 0 0 0 0]
[0 0 0 0 0]
[0 0 0 0 0]
```

CC:  @orlitzky @dcoudert @tscrim

Component: **linear algebra**

_Issue created by migration from https://trac.sagemath.org/ticket/33380_

**Comment by SandwichGouda** [comment]
Hello ! 
I am motivated to tackle this issue.
Can I be assigned to it ?

**Comment by mkoeppe** [comment]
Fixed in #37721.
